be/src/runtime/query_statistics.cpp

```diff
@@ -33,16 +35,17 @@ void QueryStatisticsRecvr::insert(const PQueryStatistics& statistics, int sender
     auto iter = _query_statistics.find(sender_id);
     if (iter == _query_statistics.end()) {
         query_statistics = new QueryStatistics;
+        auto cleanup = MakeScopedCleanup([&]() { delete query_statistics; });
         _query_statistics[sender_id] = query_statistics;
+        cleanup.cancel();
     } else {
         query_statistics = iter->second;
     }
     query_statistics->merge_pb(statistics);
 }
 
 QueryStatisticsRecvr::~QueryStatisticsRecvr() {
-    // It is unnecessary to lock here, because the destructor will be
-    // called alter DataStreamRecvr's close in ExchangeNode.
+    std::lock_guard<SpinLock> l(_lock);
```

Comment:
Why add the lock

Reply:
`_query_statistics` should be protected by lock.

Reply:
the comment is: It is unnecessary to lock here